mars/services/scheduling/supervisor/globalslot.py

```diff
@@ -90,3 +95,38 @@ def release_subtask_slots(self, band: Tuple, session_id: str, subtask_id: str):
 
     def get_used_slots(self):
         return self._band_used_slots
+
+    async def get_available_bands(self):
+        if not self._band_total_slots:
+            self._band_total_slots = await self._cluster_api.get_all_bands()
+
+        def exclude_bands(all_bands_slots, excluded_bands):
+            return {x: all_bands_slots[x] for x in all_bands_slots if x not in excluded_bands}
+        return exclude_bands(self._band_total_slots, self._blocked_bands)
+
+    async def watch_available_bands(self):
+        async def waiter():
+            try:
+                await self._blocklist_event.wait()
```

Comment:
I'm not sure whether reusing same lock will cause hang or some notification miss